backend/LexCore/Entities/MediaFile.cs

```diff
@@ -0,0 +1,98 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace LexCore.Entities;
+
+public class MediaFile : EntityBase
+{
+    // Id property comes from EntityBase
+    public required string Filename { get; set; }
+    public Guid ProjectId { get; set; } // required; all files must belong to a project
+    public DateTimeOffset LastModified { get; set; } // You know what? We *should* derive from EntityBase after all.
+    public FileMetadata? Metadata { get; set; } = new FileMetadata();
+
+    [MemberNotNull(nameof(Metadata))]
+    public void InitializeMetadataIfNeeded(string filePath)
+    {
+        if (Metadata is null)
+        {
+            var fileInfo = new FileInfo(filePath);
+            int cappedSize = 0;
+            if (fileInfo.Exists)
+            {
+                cappedSize = fileInfo.Length > int.MaxValue ? int.MaxValue : (int)fileInfo.Length;
+            }
+            Metadata = new FileMetadata
+            {
+                SizeInBytes = cappedSize,
+            };
+        }
+    }
+}
+
+public class FileMetadata
+{
+    public string? Sha256Hash { get; set; } // Used for EntityTag / ETag headers, among other things
+    public int? SizeInBytes { get; set; }
+    public string? FileFormat { get; set; } // TODO: Define strings we might use here, or else switch to MimeType
+    public string? MimeType { get; set; }
+    public string? Author { get; set; }
+    public DateTimeOffset? UploadDate { get; set; }
+    public MediaFileLicense? License { get; set; }
+
+    // Other optional, not-yet-mapped properties like purpose, transcript, etc., should end up in here
+    [JsonExtensionData]
+    public IDictionary<string, object> ExtraFields { get; set; } = new Dictionary<string, object>();
```

Comment:
I thought we were going to use `Dictionary<string, string>` here? object makes this potentially much harder to use and it can be a JsonElement in some cases.

Reply:
Turns out if it's JsonElement it's very hard to get the data into ExtraFields "by hand" in the HandleFileUpload method, and `IDictionary<string, string>` is explicitly disallowed by the JsonExtensionData attribute (sigh).

Reply:
Done, I think? Let me know if you still want me to change this, but I think any changes should probably be pushed to a later PR to avoid even more scope creep.

Reply:
yeah this is fine